src/lib/ai/emotions/EmotionSynthesizer.ts

```
@@ -31,31 +58,81 @@ export class EmotionSynthesizer {
+      for (const key in newEmotions) {
+        newEmotions[key] = (newEmotions[key] ?? 0) * decayFactor;
+        // Add slight random fluctuation to make it feel more alive, keep it small
+        if (randomFluctuation > 0) {
+            const noise = (Math.random() - 0.5) * 2 * randomFluctuation; // between -randomFluctuation and +randomFluctuation
+            newEmotions[key] = Math.max(0, Math.min(1, (newEmotions[key] ?? 0) + noise));
+        }
+      }
+
+      // 2. Apply baseEmotion influence (if provided)
+      // This is like the old 'targetEmotion' but blends more smoothly.
+      if (baseEmotion && newEmotions.hasOwnProperty(baseEmotion)) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)